lib/elixir/lib/enum.ex

```diff
@@ -1737,7 +1737,12 @@ defmodule Enum do
   end
 
   def reverse(enumerable, tail) do
-    reduce(enumerable, to_list(tail), fn(entry, acc) ->
+    do_reverse(enumerable, to_list(tail))
+  end
+
+  @spec do_reverse(t, list) :: list
```

Comment:
well, to be honest it is not. and that's why I added a spec.
`revert(enumerable, tail)` accepts `(t, t)`,
but `do_revert(enumerable, tail)` accepts `(t, list)`

in addition to that the second argument for enum reduce can be `any`,